### ISSUE TITLE
[CSR-791] cypress-cloud is not compatible with cypress 13.x

### ISSUE THREAD

**Issue by anlambert** [currents-dev/cypress-currents-reporter#179]
### Before opening, please confirm:

- [X] I have searched for [duplicate or closed issues](https://github.com/currents-dev/cypress-cloud/issues) and [discussions](https://github.com/currents-dev/cypress-cloud/discussions).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I acknowledge that I will attach a **full debug log**, otherwise the issue will be closed with no response.

### Environment information

<details>

```
# Put output below this line

System:
    OS: Linux 6.1 Debian GNU/Linux 12 (bookworm) 12 (bookworm)
    CPU: (8) x64 Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz
    Memory: 18.94 GB / 31.16 GB
    Container: Yes
    Shell: 5.2.15 - /bin/bash
  Binaries:
    Node: 18.16.0 - /usr/bin/node
    Yarn: 1.22.19 - /usr/bin/yarn
    npm: 9.5.1 - /usr/bin/npm
  Browsers:
    Chrome: 114.0.5735.90
    Chromium: 116.0.5845.140
  npmPackages:
    eslint-config-custom: latest => 0.0.0 
    prettier: latest => 3.0.2 
    turbo: ^1.10.13 => 1.10.13 
  npmGlobalPackages:
    azurite: 3.23.0
    corepack: 0.17.0
    gulp-cli: 2.3.0
    npm: 9.5.1
    samfirm: 0.2.0

```

</details>

### Describe the bug

Due to this [cypress commit](https://github.com/cypress-io/cypress/commit/069d3b399c53b02bd1206d9923c469369475da35) that introduces some breaking API changes in cypress 13.x, I noticed two issues when using cypress-cloud with it:

1.  Executed cypress tests are always reported as failing whatever their real statuses are.

See sample cypress-cloud output put below:
<details>

```
$ yarn run cypress-cloud run --parallel --record --key somekey --cloud-debug true --ci-build-id hello-cypress36
yarn run v1.22.19
$ /home/anlambert/swh/swh-environment/swh-web/node_modules/.bin/cypress-cloud run --parallel --record --key somekey --cloud-debug true --ci-build-id hello-cypress36
 WARNING  Failed to load config file, falling back to the default config. Attempted locations: [
  'file:///home/anlambert/swh/swh-environment/swh-web/currents.config.js',
  'file:///home/anlambert/swh/swh-environment/swh-web/currents.config.cjs',
  'file:///home/anlambert/swh/swh-environment/swh-web/currents.config.mjs'
]

Copyright (C) 2023 Currents Software Inc https://currents.dev
This is free software, and you are welcome to redistribute it under certain
conditions. This program comes with no warranty. Parts of this program are MIT
licensed. Refer to the license for details
https://github.com/currents-dev/cypress-cloud/blob/main/LICENSE.md

Discovered 1 spec files
Tags: false; Group: false; Parallel: true; Batch Size: 3
Connecting to cloud orchestration service...
🎥 Run URL: http://localhost:8080/run/aff6e1ec70f400b22c4deeb5b83332e7

====================================================================================================

Running: cypress/e2e/home.cy.js (1/1)

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        13.1.0                                                                         │
  │ Browser:        Electron 106 (headless)                                                        │
  │ Node Version:   v18.16.0 (/usr/bin/node)                                                       │
  │ Specs:          1 found (home.cy.js)                                                           │
  │ Searched:       cypress/e2e/home.cy.js                                                         │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  home.cy.js                                                                      (1 of 1)


  Home Page Tests
    ✓ should have focus on search form after page load
    ✓ should display positive stats for each category
    ✓ should display null counters and hide history graphs when storage is empty
    ✓ should hide counters when data is missing
    ✓ should redirect to search page when submitting search form


  5 passing (6s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        5                                                                                │
  │ Passing:      5                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     6 seconds                                                                        │
  │ Spec Ran:     home.cy.js                                                                       │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  home.cy.js                               00:06        5        5        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        00:06        5        5        -        -        -  


  Reporting results and artifacts in background...  


====================================================================================================


  Cloud Run Finished  

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │     Spec                                         Tests   Passing   Failing   Pending   Skipped │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖   cypress/e2e/home.cy.js               0ms         1         -         1         -         - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖   1 of 1 failed                        0ms         1         -         1         -         - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘

🏁 Recorded Run: http://localhost:8080/run/aff6e1ec70f400b22c4deeb5b83332e7

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
</details>

After reading cypress-cloud code, it seems that issue can be fixed using the following diff:
```diff
diff --git a/packages/cypress-cloud/lib/results/results.ts b/packages/cypress-cloud/lib/results/results.ts
index f9380de..635f0b4 100644
--- a/packages/cypress-cloud/lib/results/results.ts
+++ b/packages/cypress-cloud/lib/results/results.ts
@@ -12,11 +12,6 @@ import { ConfigState } from "../state";
 
 const debug = Debug("currents:results");
 
-export const isSuccessResult = (
-  result: CypressResult
-): result is CypressCommandLine.CypressRunResult => {
-  return result.status === "finished";
-};
 
 export const getScreenshotsSummary = (
   tests: CypressCommandLine.TestResult[] = []
@@ -282,9 +277,6 @@ export function getCypressRunResultForSpec(
   spec: string,
   cypressResult: CypressResult
 ): CypressCommandLine.CypressRunResult | undefined {
-  if (!isSuccessResult(cypressResult)) {
-    return;
-  }
 
   const run = cypressResult.runs.find((r) => r.spec.relative === spec);
   if (!run) {
```

2. Screenshots fail to be processed.

Once the fix above applied, there is an error when cypress-cloud attempts to process test screenshots, see output below:

<details>

```
$ yarn run cypress-cloud run --parallel --record --key somekey --cloud-debug true --ci-build-id hello-cypress38
yarn run v1.22.19
$ /home/anlambert/swh/swh-environment/swh-web/node_modules/.bin/cypress-cloud run --parallel --record --key somekey --cloud-debug true --ci-build-id hello-cypress38
 WARNING  Failed to load config file, falling back to the default config. Attempted locations: [
  'file:///home/anlambert/swh/swh-environment/swh-web/currents.config.js',
  'file:///home/anlambert/swh/swh-environment/swh-web/currents.config.cjs',
  'file:///home/anlambert/swh/swh-environment/swh-web/currents.config.mjs'
]

Copyright (C) 2023 Currents Software Inc https://currents.dev
This is free software, and you are welcome to redistribute it under certain
conditions. This program comes with no warranty. Parts of this program are MIT
licensed. Refer to the license for details
https://github.com/currents-dev/cypress-cloud/blob/main/LICENSE.md

Discovered 1 spec files
Tags: false; Group: false; Parallel: true; Batch Size: 3
Connecting to cloud orchestration service...
🎥 Run URL: http://localhost:8080/run/8daacfecbd761c3d663e7e4bd2646eae

====================================================================================================

Running: cypress/e2e/home.cy.js (1/1)

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        13.1.0                                                                         │
  │ Browser:        Electron 106 (headless)                                                        │
  │ Node Version:   v18.16.0 (/usr/bin/node)                                                       │
  │ Specs:          1 found (home.cy.js)                                                           │
  │ Searched:       cypress/e2e/home.cy.js                                                         │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  home.cy.js                                                                      (1 of 1)


  Home Page Tests
    ✓ should have focus on search form after page load
    ✓ should display positive stats for each category
    ✓ should display null counters and hide history graphs when storage is empty
    ✓ should hide counters when data is missing
    ✓ should redirect to search page when submitting search form


  5 passing (6s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        5                                                                                │
  │ Passing:      5                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     6 seconds                                                                        │
  │ Spec Ran:     home.cy.js                                                                       │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  home.cy.js                               00:06        5        5        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        00:06        5        5        -        -        -  


  Reporting results and artifacts in background...  

 ERROR  TypeError: Cannot read properties of undefined (reading 'flatMap')
    at /home/anlambert/swh/swh-environment/swh-web/node_modules/lib/results/results.ts:26:21
    at Array.flatMap (<anonymous>)
    at /home/anlambert/swh/swh-environment/swh-web/node_modules/lib/results/results.ts:25:19
    at Array.flatMap (<anonymous>)
    at getScreenshotsSummary (/home/anlambert/swh/swh-environment/swh-web/node_modules/lib/results/results.ts:24:16)
    at getInstanceResultPayload (/home/anlambert/swh/swh-environment/swh-web/node_modules/lib/results/results.ts:62:18)
    at getReportResultsTask (/home/anlambert/swh/swh-environment/swh-web/node_modules/lib/results/uploadResults.ts:22:13)
    at createReportTask (/home/anlambert/swh/swh-environment/swh-web/node_modules/lib/runner/reportTask.ts:26:3)
    at /home/anlambert/swh/swh-environment/swh-web/node_modules/lib/runner/runner.ts:55:7
    at Array.forEach (<anonymous>)
    at runTillDone (/home/anlambert/swh/swh-environment/swh-web/node_modules/lib/runner/runner.ts:54:7)


====================================================================================================


  Cloud Run Finished  

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │     Spec                                         Tests   Passing   Failing   Pending   Skipped │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔   undefined                           6.2s         5         5         -         -         - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔   All specs passed!                   6.2s         5         5         -         -         - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘

🏁 Recorded Run: http://localhost:8080/run/8daacfecbd761c3d663e7e4bd2646eae

Done in 13.82s.
```

</details>

The fix for this one is not straightforward to me.

### Expected behavior

cypress 13.x should be supported.

### Command and Setup

Just run any cypress-cloud session with cypress 13.x and the reported errors will show up.

### Full log and debug output

[cypress_cloud_debug_logs.txt](https://github.com/currents-dev/cypress-cloud/files/12498373/cypress_cloud_debug_logs.txt)

**Comment by agoldis** [comment]
THe fix is in progress

**Comment by Roemer** [comment]
Is this also related to the issue that viewing a run in Sorry Cypress is not possible and shows the following error:
![image](https://github.com/currents-dev/cypress-cloud/assets/393641/f42b42f3-c5cc-4e04-89ca-c8684abbe663)

**Comment by agoldis** [comment]
Please check out upgrade guide: https://currents.dev/readme/integration-with-cypress/cypress-cloud/migration-to-cypress-13

**Comment by anlambert** [comment]
I just tested the upgrade to cypress-cloud beta and tests are no longer reported as failing, see https://jenkins.softwareheritage.org/blue/organizations/jenkins/DWAPPS%2Fgitlab-builds/detail/gitlab-builds/302/pipeline/89/.

There is still the error about screenshots that appears after each test suite though but if you do not use that cypress feature it is not critical for running the tests.

**Comment by agoldis** [comment]
Thanks @anlambert and @richardmward, working on the fixes for those issues

**Comment by estefafdez** [comment]
For us, after updating everything to the latest version we got the following error, and the specs are not executed: 

```
11:47:24  Copyright (C) 2023 Currents Software Inc https://currents.dev
11:47:24  This is free software, and you are welcome to redistribute it under certain
11:47:24  conditions. This program comes with no warranty. Parts of this program are MIT
11:47:24  licensed. Refer to the license for details
11:47:24  https://github.com/currents-dev/cypress-cloud/blob/main/LICENSE.md
11:47:24  
11:47:26  
11:47:26  
11:47:26    (Attempt 1 of 2) An uncaught error was detected outside of a test
11:47:26    1) An uncaught error was detected outside of a test
11:47:26  
11:47:26    0 passing (910ms)
11:47:26    1 failing
11:47:26  
11:47:26    1) An uncaught error was detected outside of a test:
11:47:26       Error: The following error originated from your test code, not from Cypress.
11:47:26  
11:47:26    > Dynamic require of "@neuralegion/cypress-har-generator/commands" is not supported
11:47:26  
11:47:26  When Cypress detects uncaught errors originating from your test code it will automatically fail the current test.
11:47:26  
11:47:26  Cypress could not associate this error to any specific test.
11:47:26  
```

Versions: 
 - "cypress": "13.1.0",
 - "cypress-cloud": "1.9.4",
 - "cypress-debugger": "^1.0.9",

**Comment by estefafdez** [comment]
The tests are working with 

```
 "cypress": "13.1.0",
    "cypress-cloud": "1.10.0-beta.2",
    "cypress-debugger": "1.0.7",`
``` 

and you can see the logs on the command line that the tests passed ok: 

``` 
12:45:25    (Run Starting)
12:45:25  
12:45:25    ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
12:45:25    │ Cypress:        13.1.0                                                                         │
12:45:25    │ Browser:        Electron 106 (headless)                                                        │
12:45:25    │ Node Version:   v16.16.0 (/usr/bin/node)                                                       │
12:45:25    │ Specs:          1 found (admin_auth.spec.ts)                                                   │
12:45:25    │ Searched:       cypress/test/genially/admin/admin_auth.spec.ts                                 │
12:45:25    │ Experiments:    experimentalRunAllSpecs=true,experimentalMemoryManagement=true,experimentalWe… │
12:45:25    └────────────────────────────────────────────────────────────────────────────────────────────────┘
12:45:25  
12:45:25  
12:45:25  ────────────────────────────────────────────────────────────────────────────────────────────────────
12:45:25                                                                                                      
12:45:25    Running:  admin_auth.spec.ts                                                              (1 of 1)
12:45:32  
12:45:32  
12:45:32    Admin/Auth/AdminUser
12:45:37      ✓ should login as authorized master user (5497ms)
12:45:40      ✓ should logout as authorized master user (2265ms)
12:45:40  
12:45:40  
12:45:40    2 passing (9s)
12:45:40  
12:45:40  
12:45:40    (Results)
12:45:40  
12:45:40    ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
12:45:40    │ Tests:        2                                                                                │
12:45:40    │ Passing:      2                                                                                │
12:45:40    │ Failing:      0                                                                                │
12:45:40    │ Pending:      0                                                                                │
12:45:40    │ Skipped:      0                                                                                │
12:45:40    │ Screenshots:  0                                                                                │
12:45:40    │ Video:        false                                                                            │
12:45:40    │ Duration:     8 seconds                                                                        │
12:45:40    │ Spec Ran:     admin_auth.spec.ts                                                               │
12:45:40    └────────────────────────────────────────────────────────────────────────────────────────────────┘
12:45:40  
12:45:40  
12:45:40  resize:  can't open terminal /dev/tty
12:45:40  ====================================================================================================
12:45:40  
12:45:40    (Run Finished)
12:45:40  
12:45:40  
12:45:40         Spec                                              Tests  Passing  Failing  Pending  Skipped  
12:45:40    ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
12:45:40    │ ✔  admin_auth.spec.ts                       00:08        2        2        -        -        - │
12:45:40    └────────────────────────────────────────────────────────────────────────────────────────────────┘
12:45:40      ✔  All specs passed!                        00:08        2        2        -        -        -  
12:45:40  
12:45:40  
12:45:40    Reporting results and artifacts in background...  
12:45:40  
12:45:40   ERROR  TypeError: Cannot read properties of undefined (reading 'flatMap')
12:45:40      at /home/ubuntu/workspace/workspace/ressED_qa_PD-13026_cy13_recorder/mono-genially/qa/e2e/node_modules/lib/results/results.ts:27:21
12:45:40      at Array.flatMap (<anonymous>)
12:45:40      at /home/ubuntu/workspace/workspace/ressED_qa_PD-13026_cy13_recorder/mono-genially/qa/e2e/node_modules/lib/results/results.ts:26:19
12:45:40      at Array.flatMap (<anonymous>)
12:45:40      at getRunScreenshots (/home/ubuntu/workspace/workspace/ressED_qa_PD-13026_cy13_recorder/mono-genially/qa/e2e/node_modules/lib/results/results.ts:25:16)
12:45:40      at getInstanceResultPayload (/home/ubuntu/workspace/workspace/ressED_qa_PD-13026_cy13_recorder/mono-genially/qa/e2e/node_modules/lib/results/results.ts:66:18)
12:45:40      at getReportResultsTask (/home/ubuntu/workspace/workspace/ressED_qa_PD-13026_cy13_recorder/mono-genially/qa/e2e/node_modules/lib/results/uploadResults.ts:26:27)
12:45:40      at createReportTask (/home/ubuntu/workspace/workspace/ressED_qa_PD-13026_cy13_recorder/mono-genially/qa/e2e/node_modules/lib/runner/reportTask.ts:30:5)
12:45:40      at /home/ubuntu/workspace/workspace/ressED_qa_PD-13026_cy13_recorder/mono-genially/qa/e2e/node_modules/lib/runner/runner.ts:58:7
12:45:40      at Array.forEach (<anonymous>)
12:45:40      at runTillDone (/home/ubuntu/workspace/workspace/ressED_qa_PD-13026_cy13_recorder/mono-genially/qa/e2e/node_modules/lib/runner/runner.ts:57:14)
12:45:40  
12:45:40  
12:45:40  ====================================================================================================
12:45:40  
12:45:40  
12:45:40    Cloud Run Finished  
12:45:40  
12:45:40    ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
12:45:40    │     Spec                                         Tests   Passing   Failing   Pending   Skipped │
12:45:40    ├────────────────────────────────────────────────────────────────────────────────────────────────┤
12:45:40    │ ✔   cypress/test/genially/admin/ad      8.6s         2         2         -         -         - │
12:45:40    │     min_auth.spec.ts                                                                           │
12:45:40    ├────────────────────────────────────────────────────────────────────────────────────────────────┤
12:45:40    │ ✔   All specs passed!                   8.6s         2         2         -         -         - │
12:45:40    └────────────────────────────────────────────────────────────────────────────────────────────────┘
12:45:40  
12:45:40  🏁 Recorded Run: https://dashboard-sorrycypress.genially.com/run/4acaa09f2aa7f36f26d5acbec86c675e
```

The problem is, that the results do not appears on sorry cypress (or the mongodb) and you get 0 tests executed (and the specs runs indefinitely) 

<img width="1673" alt="image" src="https://github.com/currents-dev/cypress-cloud/assets/6990209/fec8d3cf-3348-474b-ad49-756700e02436">

<img width="1133" alt="image" src="https://github.com/currents-dev/cypress-cloud/assets/6990209/79641042-133c-4f1c-a891-dab36a0475e1">

**Comment by agoldis** [comment]
@estefafdez thanks for reporting - will keep cypress-debugger issue separated

@Roemer, @anlambert and @richardmward please check out `1.10.0-beta.3` also, please submit the [debug logs](https://currents.dev/readme/integration-with-cypress/cypress-cloud/migration-to-cypress-13) when you face issues - would be much easier for us, and thanks for helping with polishing the integration!

**Comment by richardmward** [comment]
That has sorted my issue. Thank you.

**Comment by estefafdez** [comment]
@agoldis the debugger issue is already reported there https://github.com/currents-dev/cypress-debugger/issues/59 The problem is that you can run the tests in Cy13 but you can't see the results in sorry cypress, Something seems to be missing as well.

**Comment by andreiho** [comment]
Anyone else experiencing tests just getting stuck and never finishing? I'm using `cypress@13.1.0` with `cypress-cloud@1.10.0-beta.3`, having followed the [migration guide](https://currents.dev/readme/integration-with-cypress/cypress-cloud/migration-to-cypress-13), and the tests start, but it just hangs after the first one. This happens both in `open` and `run` mode, and if I remove the `cypress-cloud/support` and `cypress-cloud/plugin` parts, the tests work. Any idea what is going on?

**Comment by estefafdez** [comment]
@anlambert what did you see on the cypress dashboard when the run is finished? did you see the results there?

**Comment by agoldis** [comment]
@estefafdez @richardmward Please check out `1.10.0-beta.4` - it has a few additional fixes. 

Also, the corrupted runs could have incompatible data, they should be removed from the DB.

**Comment by anlambert** [comment]
@estefafdez  regarding the sorry-cypress dashboard feature, I bumped cypress-cloud to `1.10.0-beta.4` and tested it locally using that [docker-compose.yml](https://github.com/sorry-cypress/sorry-cypress/blob/master/docker-compose.full.yml) file to run the sorry-cypress kit. Everything seems to work as expected, see screenshots below:

![image](https://github.com/currents-dev/cypress-cloud/assets/5493543/2f437eca-c96a-46c7-8149-d18b7c7df403)

![image](https://github.com/currents-dev/cypress-cloud/assets/5493543/973a2951-a57c-4ed1-9e5c-66591e569046)

**Comment by bilarallen** [comment]
> Anyone else experiencing tests just getting stuck and never finishing? I'm using `cypress@13.1.0` with `cypress-cloud@1.10.0-beta.3`, having followed the [migration guide](https://currents.dev/readme/integration-with-cypress/cypress-cloud/migration-to-cypress-13), and the tests start, but it just hangs after the first one. This happens both in `open` and `run` mode, and if I remove the `cypress-cloud/support` and `cypress-cloud/plugin` parts, the tests work. Any idea what is going on?

@andreiho I also experience this with `npx cypress open` The test hangs whenever I put `import "cypress-cloud/support"` in my e2e.js file but if I remove it there will be no problem with cypress runner.
![image](https://github.com/currents-dev/cypress-cloud/assets/65072341/ff8a850f-3509-417d-b288-62f0655e4382)

One thing I also noticed is that whenever I check the `cypress-cloud/support` its index.d.ts file contains an empty import
![image](https://github.com/currents-dev/cypress-cloud/assets/65072341/e1cf2fe4-bdd3-4975-bbb9-6dedd39b9961)

here's additional info:
cypress version: 13.1.0
cypress-cloud version: 1.10.0-beta.4
node version: 20.5.0

**Comment by agoldis** [comment]
@bilarallen thanks for elaborating on the issue. If that is reproducible, please share a standalone example or and debug logs (https://currents.dev/readme/integration-with-cypress/troubleshooting)

**Comment by bilarallen** [comment]
> @bilarallen thanks for elaborating on the issue. If that is reproducible, please share a standalone example or and debug logs (https://currents.dev/readme/integration-with-cypress/troubleshooting)

Hi @agoldis This can only be replicated when running a test via cypress runner through `npx cypress open` command, here's my envinfo and I'll also include steps to replicate since this only be replicated when using cypress runner

```
System:
    OS: Linux 5.10 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
    CPU: (16) x64 AMD Ryzen 7 5800U with Radeon Graphics
    Memory: 4.06 GB / 6.71 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.5.0 - ~/.nvm/versions/node/v20.5.0/bin/node
    npm: 9.8.0 - ~/.nvm/versions/node/v20.5.0/bin/npm
  Browsers:
    Chrome: 111.0.5563.64
  npmPackages:
    @babel/eslint-parser: ^7.21.8 => 7.21.8 
    @currents/cli: ^4.0.3 => 4.0.3 
    @cypress/angular:  0.0.0-development 
    @cypress/mount-utils:  0.0.0-development 
    @cypress/react:  0.0.0-development 
    @cypress/react18:  0.0.0-development 
    @cypress/svelte:  0.0.0-development 
    @cypress/vue:  0.0.0-development 
    @cypress/vue2:  0.0.0-development 
    @cypress/webpack-batteries-included-preprocessor: ^3.0.2 => 3.0.2 
    @cypress/webpack-preprocessor: ^5.17.0 => 5.17.0 
    @faker-js/faker: ^7.6.0 => 7.6.0 
    babel-eslint: ^10.1.0 => 10.1.0 
    cy-verify-downloads: ^0.1.16 => 0.1.16 
    cypress: ^13.1.0 => 13.1.0 
    cypress-cloud: ^1.10.0-beta.4 => 1.10.0-beta.4 
    cypress-delete-downloads-folder: 0.0.3 => 0.0.3 
    cypress-network-idle: ^1.14.2 => 1.14.2 
    cypress-real-events: ^1.10.1 => 1.10.1 
    cypress-recurse: ^1.27.0 => 1.27.0 
    eslint: ^8.39.0 => 8.39.0 
    moment: ^2.29.4 => 2.29.4 
    moment-timezone: ^0.5.43 => 0.5.43 
    numeral: ^2.0.6 => 2.0.6 
    swagger-client: ^3.18.5 => 3.18.5 
    uuid: ^9.0.0 => 9.0.0 (8.3.2)
    xvfb: ^0.4.0 => 0.4.0 
  npmGlobalPackages:
    corepack: 0.19.0
    npm: 9.8.0
```

Steps to replicate when using cypress runner and importing `cypress-cloud/support` in e2e.js file:
```
1. Install and use cypress version 12.3.1
2. Install and use cypress-cloud version 1.10.0-beta.4
3. Go to cypress > support > e2e.js file
4. import `cypress-cloud/support` in e2e.js file
5. Run a test using cypress runner via `npx cypress open` command and select electron browser
6. Wait for test execution to finish and observe if it hang
7. Remove import `cypress-cloud/support` in e2e.js file and repeat steps 5 to 6
```

**Comment by henrikvolmer** [comment]
@agoldis @bilarallen : I'm experiencing the same issue with `cypress open`. Is there any solution yet?

**Comment by estefafdez** [comment]
Ok, for me is finally working with: 

```
 "cypress": "13.1.0",
    "cypress-cloud": "1.10.0-beta.4",
    "cypress-debugger": "1.0.7",
```

![image](https://github.com/currents-dev/cypress-cloud/assets/6990209/66f00ff9-7fd5-4658-bcb2-dffdcde5c736)
![image](https://github.com/currents-dev/cypress-cloud/assets/6990209/b6ac5dde-a86e-41ee-ace3-55a122c6bcb7)

@agoldis thanks!

**Comment by andreiho** [comment]
@agoldis @bilarallen  I've been trying to create a reproducible example, but can't manage to. It appears to be related to some of the custom behaviour we have implemented to deal with some technicalities in our application. But I can't quite explain how importing the cypress-cloud code results in the tests hanging and cypress freezing completely. I'll try to give a bit more context into our situation.

We have a few custom commands which we run on most tests, in order to perform certain actions before proceeding with the test. In this case, the `navigateTo` command will wait for the page to finish loading, then wait for an element `#alerts-holder` to be present and attempt to clear any notifications using the `clickNotificationButton` command in order to prevent notifications covering up content which might need to be clicked in the test.

Now I've tried changing different things in these commands in an attempt to maybe fix some issues, but I can't quite get to the bottom of it, particularly why importing cypress-cloud ends up manifesting this way. It does however work if I simply just never call the `clickNotificationButton` command, even with cypress-cloud imported. The way the issue manifests is that the tests just freeze, along with the cypress browser, requiring a force quit of the process. It freezes so badly, that it makes the CPU on my M1 Pro go crazy.

This could of course be sub-optimal code on our part, but why would it only manifest itself this way while cypress-cloud is imported? I've tried looking at your `plugin` and `support` code, and with my limited knowledge I can't quite tell what could be causing issues there.

<details>
<summary>Our custom commands</summary>

```ts
export const navigateTo = (
  routePath = '/',
  options: { isLoggedIn?: boolean; force?: boolean } = {},
): void => {
  const url = new URL(`${Cypress.env('BASE_URL')}${routePath}`);
  const { force = false, isLoggedIn = true } = options;

  cy.url().then((href) => {
    if (force || href !== url.href) {
      cy.visit(url.href);
      cy.get('body').should('not.have.class', 'loading');

      if (isLoggedIn) {
        cy.get('#alerts-holder', { timeout: 20000 }).should('exist');
        cy.clickNotificationButton();
      }
    }
  });
};

export const clickNotificationButton = (
  buttonText?: string,
): Cypress.Chainable<JQuery<HTMLElement>> => {
  return cy.get('#alerts-holder').then(($aw) => {
    if (buttonText) {
      cy.get('#alerts-holder').contains(button, buttonText).click();
    } else if ($aw.find('[aria-label="close"]').length) {
      cy.safeClick(() => cy.get('#alerts-holder button[aria-label="close"]'), {
        multiple: true,
      });
    }
  });
};

export const safeClick = (
  cySelectorFn: () => Cypress.Chainable<JQuery<HTMLElement>>,
  clickOpts?: Partial<Cypress.ClickOptions>,
): Cypress.Chainable<JQuery<HTMLElement>> => {
  return cy
    .waitUntil(
      () => {
        return cySelectorFn()
          .as('clickable')
          .wait(10) // https://github.com/cypress-io/cypress/issues/7306
          .then(($el) => Cypress.dom.isAttached($el));
      },
      { timeout: 1000, interval: 10 },
    )
    .get('@clickable')
    .click(clickOpts);
};
```
</details>

**Comment by andreiho** [comment]
Okay after poking around a bit more, I think I may have found the culprit. You can probably disregard most of what I said in my previous message, although I still think there's some relation between the how tests are written, and what happens further down.

The `cypress-cloud/support` file which you can see [here](https://www.npmjs.com/package/cypress-cloud/v/1.10.0-beta.4?activeTab=code) has this piece of code attempting to stringify the entire test object using [fast-safe-stringify](https://github.com/davidmarkclements/fast-safe-stringify). Without this `safeStringify` call, the tests run perfectly. I haven't looked deeper into what exactly happens, but the full `test` object looks like a fairly complex thing which the lib might struggle with, possibly resulting in a recursive loop of sorts and blocking everything else from executing.

```ts
cy.task(
  `currents:test:after:run`,
  safeStringify({
    ...test,
    fullTitle: test.fullTitle()
  }),
  {
    log: false
  }
);
```

<details>
<summary>Screenshot of the `test` object in my case</summary>

<img width="1089" alt="Screenshot 2023-09-12 at 14 50 37" src="https://github.com/currents-dev/cypress-cloud/assets/9000674/7fcd97c4-ebfb-4a4e-859a-f703a52124bd">
</details>

**Comment by agoldis** [comment]
@andreiho thanks a lot for the detailed response. Any chance you can share the test / sample with me somehow? Another approach would be to put a breakpoint right before stringifying the result and make sure that the function returns.

**Comment by andreiho** [comment]
@agoldis I did try a breakpoint and calling the stringify function in the console, and it never returns, it just hangs indefinitely and the browser becomes virtually unusable, requiring force quit.

Haven't been able to put something useful together to share, as it seems somewhat related to some of the more complex orchestration of our application, with large number of requests made on load and various mocks and intercepts. The test itself is as simple as it gets, I tried running it on a dummy HTML webpage and everything works fine, which makes me think it's something else in the background that is part of the test object which the stringifier doesn't like. I might try manually inspecting that test object before/after commenting out some of the code I previously mentioned which seems to unblock the tests, and see what the differences are.

**Comment by agoldis** [comment]
@andreiho would you be able to join a quick support call?

**Comment by andreiho** [comment]
> @andreiho would you be able to join a quick support call? 

@agoldis I'm not able right now, it's late here in Europe, but I'll try to do a bit more digging tomorrow at work and maybe see if I can somehow give you access to something reproducible.

**Comment by andreiho** [comment]
@agoldis I went through the `test` object and tried running `safeStringify` on each property, to see where it chokes. The 2 things that fail are `commands` and `parent` (which circularly references the same test object). 


A few examples of things that fail. In these cases I specifically went through all the properties and the stringifier only freezes on these exact properties, there are probably more scenarios however. To me it just seems that the unpredictability and complexity of this `test` object is maybe too much for the stringifier library, and maybe another solution should be considered.

<details>
<summary>1. Commands that have the `$el` property referencing a DOM element</summary>

![Screenshot 2023-09-13 at 11 17 35](https://github.com/currents-dev/cypress-cloud/assets/9000674/f3aa1a90-4dea-4f13-b809-fbc25ffb42e0)
</details>

<details>
<summary>2. Intercept commands that use a regex matcher (`hostname` in this case)</summary>

![Screenshot 2023-09-13 at 11 14 24](https://github.com/currents-dev/cypress-cloud/assets/9000674/333316b2-99b7-4921-b2ef-0a9b66bfd7f3)
</details>

Anyhow, I created a super simple [repro example](https://github.com/andreiho/cypress-cloud-freeze-repro), where I load the website of a tool we use internally, which happens to make tons of requests on load. The test itself is nothing but just loading the page and asserting a button is there. Run this test with `cypress open` and it should freeze (for some reason it works with `run`, although in my application it doesn't). If you add a breakpoint in the `support/index.js` file before stringifying and you try to manually stringify the `test.commands` array, it should freeze and never return.

**Comment by agoldis** [comment]
@andreiho that's super-helpful! Will release a fix today

**Comment by agoldis** [comment]
@andreiho @henrikvolmer @bilarallen Please check out 1.10.0-beta.5 - it has a few fixes 🤞🏻

**Comment by bilarallen** [comment]
> @andreiho @henrikvolmer @bilarallen Please check out 1.10.0-beta.5 - it has a few fixes 🤞🏻

It's working now on my end having the `import "cypress-cloud/support";` in my e2e.js 👍 
![image](https://github.com/currents-dev/cypress-cloud/assets/65072341/67d30096-1a18-40d9-a4d0-1c263dcd19a5)

**Comment by andreiho** [comment]
@agoldis Everything working as expected on my end as well, great work! 🙌 Any ETA on when this will land in a proper release?

**Comment by agoldis** [comment]
@andreiho adding some tests and fixing TS errors - early next week

**Comment by estefafdez** [comment]
Working on our side as well, thank you very much @agoldis and the team for your hard work and fast speed to resolve this issue! :)

**Comment by estefafdez** [comment]
One last thing, the images are not being displayed on sorry-cypress dashboard: 

<img width="1460" alt="Screenshot 2023-09-14 at 13 50 33" src="https://github.com/currents-dev/cypress-cloud/assets/6990209/54544194-ead9-4acb-949f-97b463daa5d5">

I ran the tests with the debug mode , the logs are attached for you to have a look :) 

[logs.txt](https://github.com/currents-dev/cypress-cloud/files/12608403/logs.txt)

**Comment by agoldis** [comment]
@midleman @estefafdez @andreiho @bilarallen @henrikvolme @anlambert @Roemer

We've released a few fixes as `2.0.0-beta.0. Would appreciate your help with checking the newest version. 
It is still released in `beta` npm channel: `npm i cypress-cloud@beta`.

Please note that the major version bump and breaking changes:
- dropping support for NodeJS 14 (EOL)
- the TS definition of `run` function (`import { run } from `cypress-cloud`) was changed to support multiple versions of cypress runners

**Comment by midleman** [comment]
@agoldis YESSS, this solves both issues previously reported:
1. hanging on running in GUI mode
2. results not being reported in Currents Dashboard

Thank you!

**Comment by bilarallen** [comment]
Great! The fix is confirmed working 😉 thanks @agoldis! 🎉

**Comment by agoldis** [comment]
@estefafdez any luck?

**Comment by estefafdez** [comment]
@agoldis working also on our side! :) thankssss

**Comment by andreiho** [comment]
All looking good here as well 👌

**Comment by grushetsky** [comment]
After upgrading to Cypress 13.2.0 and cypress-cloud 2.0.0-beta.0 the following error started to occur in the CI runner:
```
CypressError: `cy.task('currents:test:after:run')` failed with the following error:
The 'task' event has not been registered in the setupNodeEvents method. You must register it before using cy.task()
Fix this in your setupNodeEvents method here:
…/cypress.config.ts
https://on.cypress.io/api/task
Because this error occurred during a `after each` hook we are skipping all of the remaining tests.
```

The cloud plugin is called in the same way as it was previously called in _cypress.config.ts_:
```typescript
export default defineConfig({
  e2e: {
    setupNodeEvents(on, config) {
      cloudPlugin(on, config);

      on('before:browser:launch', (browser, launchOptions) => {
        …
      });

      on(
        'file:preprocessor',
        webpackPreprocessor({
          …
        })
      );
    },
  },
});
```

What is the catch here?

**Comment by midleman** [comment]
Slight update. Things are working, but I have noticed that we are only getting video uploads in Currents Dashboard if the test fails and _not_ on retries. Historically this has worked fine using their [recommended configuration](https://docs.cypress.io/guides/guides/test-retries#Only-upload-videos-for-specs-with-failing-or-retried-tests). Just curious if anyone else has the same issue or if I need to look further into my configuration.

**Comment by estefafdez** [comment]
@midleman in Cypress 13 they removed the video, so no video should be ever sent into the sorry cypress dashboard. Have you tried to do the migration properly? https://www.cypress.io/blog/2023/09/15/announcing-cypress-13-test-replay/

**Comment by anlambert** [comment]
@agoldis Still all good from my side using `cypress-cloud 2.0.0-beta.0` and `cypress 13.2.0` https://jenkins.softwareheritage.org/blue/organizations/jenkins/DWAPPS%2Fgitlab-builds/detail/gitlab-builds/315/pipeline/88

**Comment by mirobo** [comment]
> > @midleman in Cypress 13 they removed the video, so no video should be ever sent into the sorry cypress dashboard. Have you tried to do the migration properly? https://www.cypress.io/blog/2023/09/15/announcing-cypress-13-test-replay/
> 
> I followed the migration guide, which states you can manually set `video: true` (which I have) - see: [here](https://docs.cypress.io/guides/guides/test-retries#Only-upload-videos-for-specs-with-failing-or-retried-tests). The videos upload fine to Currents if the test fails, just not on retries. I confirmed the logic in the after:spec is correct as well, so I'm confused why it's not uploading in that particular case. 🤔

I have the same issue, but I wasn't able to reproduce it. I created flaky tests with retries, failing tests, all with different nestings and videos were always uploaded on retries and failures. But then in "reality" (with the same code in "after:spec" hook) I **sometimes** (!) miss videos of retried tests. I didn't investigate further. This happend with Cypress 12,x; I haven't tried 13+ yet..

**Comment by agoldis** [comment]
@midleman @mirobo @estefafdez
 - setting `video:true` enables videos in cypress 13
 - please tell me if setting `videoUploadOnPasses: true` in `cypress.config.js` (despite the warnings) solves the issue for you

**Comment by agoldis** [comment]
@grushetsky 

Please try this:

```ts
//... 
async setupNodeEvents(on, config) {
    await cloudPlugin(on, config);
    //... 
}
```

**Comment by midleman** [comment]
> * setting `video:true` enables videos in cypress 13
> * please tell me if setting `videoUploadOnPasses: true` in `cypress.config.js` (despite the warnings) solves the issue for you

ah interesting, yep, that fixes it!

**Comment by romankhomitskyi** [comment]
looks like everythins is good
passed on CI

**Comment by agoldis** [comment]
I have updated the guide for setting up cypress-cloud together with multiple plugins. There are a few caveats, please take a look here: https://github.com/currents-dev/cypress-cloud#setup-with-existing-plugins

**Comment by grushetsky** [comment]
@agoldis, the suggested fix works! Thank you!

**Comment by agoldis** [comment]
> @agoldis, the suggested fix works! Thank you!

Awesome, I also updated the documentation

**Comment by romankhomitskyi** [comment]
"cypress": "13.2.0",
"cypress-cloud": "2.0.0-beta.0",

```
async setupNodeEvents(on, config) {
cypressGrep(config);
return await cloudPlugin(on, config);
}
```

<img width="321" alt="Screenshot 2023-09-21 at 13 39 32" src="https://github.com/currents-dev/cypress-cloud/assets/57874507/ed8538c3-aff0-456f-9064-28da2a517fb3">

![image](https://github.com/currents-dev/cypress-cloud/assets/57874507/76aac383-59d0-40c3-a1a1-b5d830b361e6)


I have an error for random specs, which occurs after I switched from cy2 to cypress cloud
What can cause the  issue?

**Comment by agoldis** [comment]
@romankhomitskyi i need more data to troubleshoot - the debug logs of the affected runs (see cypress-cloud troubleshooting section) and the dump of the affected run item from the database.

**Comment by agoldis** [comment]
Released `2.0.0-beta.1` with a few fixes and improvements:
    
    - Set the default value of `videoUploadOnPasses` to `true`
    - Add more debug logs
    - Add `currents_marker` to warn about config overwrites
    - Print currents and cypress version information when starting a run

**Comment by romankhomitskyi** [comment]
> @romankhomitskyi i need more data to troubleshoot - the debug logs of the affected runs (see cypress-cloud troubleshooting section) and the dump of the affected run item from the database.

![image](https://github.com/currents-dev/cypress-cloud/assets/57874507/b0d20834-6ca5-4b89-b10d-632e468b6a5f)
<img width="1175" alt="Screenshot 2023-09-24 at 20 01 45" src="https://github.com/currents-dev/cypress-cloud/assets/57874507/55a03fef-0a70-439e-b146-9c3c8777efc8">
I guess that error causing the issue

**Comment by midleman** [comment]
ok, just noticed something, 
`2.0.0-beta.0` uploads screenshots on failures
`2.0.0-beta.1` does _not_ upload screenshots on failures

i have the same exact config between both and didn't change anything other than the `cypress-cloud` package version.

**Comment by agoldis** [comment]
@midleman is this consistent? I am challenged because we do have tests that verify uploads on failures. If you have a chance to generate a debug log, please do.

**Comment by bilarallen** [comment]
> ok, just noticed something, 
> `2.0.0-beta.0` uploads screenshots on failures 
> `2.0.0-beta.1` does _not_ upload screenshots on failures 
> i have the same exact config between both and didn't change anything other than the `cypress-cloud` package version.

@midleman @agoldis I have upgraded to `2.0.0-beta.1` and the following is the behavior I noticed with cypress-cloud:

- Confirmed that `2.0.0-beta.1` does not upload screenshots on failures and on flaky run
![image](https://github.com/currents-dev/cypress-cloud/assets/65072341/ad9281b3-48f4-4e5e-b131-45885b9db6f6)

- Aside from that the test always includes a video which was not the usual behavior before the update (I guess uploading videos for passed tests also takes time in the run? CMIIW)
![image](https://github.com/currents-dev/cypress-cloud/assets/65072341/48dddeaf-7ba7-40d8-abf1-8a842e64ef61)

For the video, I think this has something to do with `Set the default value of videoUploadOnPasses to true` that's been a default in `2.0.0-beta.1`

Before in `2.0.0-beta.0` my cypress config are the following and all works fine and I didn't change it when I upgraded to `2.0.0-beta.1`:
```
  videoCompression: false,
  video: true,
  screenshotOnRunFailure: true,
```

**Comment by agoldis** [comment]
@bilarallen @midleman 

- `videoUploadOnPasses`

The option was removed in Cypress 13 - keeping it will trigger a warning message. Enabling videos (`videos: true`) will activate videos for all tests, regardless of the outcome. We wanted to have similar behaviour to avoid confusion. You can still provide this option and cypress-cloud will respect it - it will trigger Cypress 13 warnings, though. 

- missing screenshots for failed / flaky tests

The changes between `2.0.0-beta.0` and `2.0.0-beta.1` are not functional - we added tests and a few debug statements. It's very surprising to see different behaviour. We need the debug logs to understand what caused the screenshots to disappear. After the changes to Module API output, we use more complicated methods to capture run information, including screenshots and test attempt details. But that also complicates the troubleshooting. Your help with producing the debug logs or a standalone isolated reproducible example is appreciated - the issues you're reporting are preventing us from releasing 2.0.0.

**Comment by venunh** [comment]
Hi @agoldis , Right now am using cypress 10.8.0 with cy2 to achieve parallel execution and with sorry cypress dashboard to view results. Now am planning to upgrade to cypress 13.2.0 with cypress cloud and sorry cypress dashboard. Can you please guide me how to perform this ?
I tried creating currents.config.js file and added importing cypress-cloud plugin and support in cypress.config.ts and e2e.js file and used below command to execute.
npx cypress-cloud run --parallel --record --key somekey --ci-build-id test_build_123.

I started seeing error message as ,
 Unable to resolve cypress configuration
- make sure that cypress-cloud/plugin is installed error along with
- cypress stdout: cannot find module @packages/server/lib/plugins/child/require_async_child

Below are the version used:
cypress - 13.2.0
cypress-cloud - beta

**Comment by bilarallen** [comment]
> @bilarallen @midleman
> 
> * `videoUploadOnPasses`
> 
> The option was removed in Cypress 13 - keeping it will trigger a warning message. Enabling videos (`videos: true`) will activate videos for all tests, regardless of the outcome. We wanted to have similar behaviour to avoid confusion. You can still provide this option and cypress-cloud will respect it - it will trigger Cypress 13 warnings, though.
> 
> * missing screenshots for failed / flaky tests
> 
> The changes between `2.0.0-beta.0` and `2.0.0-beta.1` are not functional - we added tests and a few debug statements. It's very surprising to see different behaviour. We need the debug logs to understand what caused the screenshots to disappear. After the changes to Module API output, we use more complicated methods to capture run information, including screenshots and test attempt details. But that also complicates the troubleshooting. Your he
lp with **producing the debug logs or a standalone isolated reproducible example is appreciated** - the issues you're reporting are preventing us from releasing 2.0.0.

Hi @agoldis I tried to run a failing test with debug logs, hope this helps.

[cypress_cloud_issues_179_no_screenshot_logs.txt](https://github.com/currents-dev/cypress-cloud/files/12725758/cypress_cloud_issues_179_no_screenshot_logs.txt)

**Comment by agoldis** [comment]
@bilarallen Thanks, that was helpful!

I can see that the log file has some missing some event that should have been emitted by cypress. 
Please try installing `cypress-on-fix` plugin as described here: https://github.com/currents-dev/cypress-cloud#event-callbacks-for-multiple-plugins and tell me if it fixed the issue.

**Comment by bilarallen** [comment]
> @bilarallen Thanks, that was helpful!
> 
> I can see that the log file has some missing some event that should have been emitted by cypress. Please try installing `cypress-on-fix` plugin as described here: https://github.com/currents-dev/cypress-cloud#event-callbacks-for-multiple-plugins and tell me if it fixed the issue.

Hi, @agoldis I tried to use the `cypress-on-fix` plugin and it worked! The screenshot in the failed/flaky run is now included, but it seems it spawned another issue wherein the screenshot didn't capture the error or the moment the test failed also there's some issue with the video timestamp this occurred in all test retries.

![image](https://github.com/currents-dev/cypress-cloud/assets/65072341/e88c35f4-789a-4fee-b0ba-1f9683dc4091)

![image](https://github.com/currents-dev/cypress-cloud/assets/65072341/4f4eeaa7-906c-4c47-a1e9-629a92d98410)

**Comment by estefafdez** [comment]
Cypress 13 got us this error tonight: 

```
20:08:25  [AWS CodeBuild Plugin] We've detected that you're using a 3rd party library that is not supported by Cypress: cli.js
20:08:25  [AWS CodeBuild Plugin] To continue running Cypress, please remove this library or reach out for help migrating.
```

it's working with cypress 12.17.4 but not working again with cypress 13 :(

**Comment by dangenendt** [comment]
@estefafdez 
Same here... i guess, cypress now checks the dependencies and throw this error when they see "cypress-cloud" package. 
Can we patch this? 🧐^^

**Comment by gmerwan** [comment]
Same as @estefafdez
Has anyone found a solution for this ?

**Comment by estefafdez** [comment]
Right now, downgrade to cypress 12 (latest version) and wait until the currents teams think about a solution :(

**Comment by dangenendt** [comment]
This will be hard one, i guess... 
Seems like this check happens partially remote. If so, its an interesting point that cypress seems to call home 🤡 
Our Pipeline already worked with cypress 13.2 since some days... so last night this check "has been activated" in some way

**Comment by estefafdez** [comment]
> This will be hard one, i guess... Seems like this check happens partially remote. If so, its an interesting point that cypress seems to call home with some repo infos 🤡 Our Pipeline already worked with cypress 13.2 since some days... so last night this check "has been activated" in some way

Same on our side...

**Comment by anlambert** [comment]
> This will be hard one, i guess...
Seems like this check happens partially remote. If so, its an interesting point that cypress seems to call home with some repo infos 🤡
Our Pipeline already worked with cypress 13.2 since some days... so last night this check "has been activated" in some way

Same issue from my side, regardless `cypress-cloud` version used (tested with `1.10.0-beta.4` and `2.0.0-beta.1`): https://jenkins.softwareheritage.org/blue/organizations/jenkins/DWAPPS%2Fgitlab-builds/detail/gitlab-builds/320/pipeline.

Curiously, I can run the tests locally without any issue:
```
$ CURRENTS_API_URL=http://localhost:1234 CURRENTS_PROJECT_ID=swh-web yarn run cypress-cloud run --record --key swh-web --parallel --ci-build-id build-2
yarn run v1.22.19
$ /home/anlambert/swh/swh-environment/swh-web/node_modules/.bin/cypress-cloud run --record --key swh-web --parallel --ci-build-id build-2
 WARNING  Failed to load config file, falling back to the default config. Attempted locations: [
  'file:///home/anlambert/swh/swh-environment/swh-web/currents.config.js',
  'file:///home/anlambert/swh/swh-environment/swh-web/currents.config.cjs',
  'file:///home/anlambert/swh/swh-environment/swh-web/currents.config.mjs'
]

Copyright (C) 2023 Currents Software Inc https://currents.dev
This is free software, and you are welcome to redistribute it under certain
conditions. This program comes with no warranty. Parts of this program are MIT
licensed. Refer to the license for details
https://github.com/currents-dev/cypress-cloud/blob/main/LICENSE.md

Cypress-cloud version: 2.0.0-beta.1
Cypress version: 13.3.0
Discovered 25 spec files
Tags: false; Group: false; Parallel: true; Batch Size: 3
Connecting to cloud orchestration service...
🎥 Run URL: http://localhost:8080/run/e455e0bdc0d3226e99bc24547f2ea7ab

====================================================================================================

Running: cypress/e2e/add-forge-now-request-dashboard.cy.js (2/25)

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        13.3.0                                                                         │
  │ Browser:        Electron 114 (headless)                                                        │
  │ Node Version:   v18.16.0 (/usr/bin/node)                                                       │
  │ Specs:          1 found (add-forge-now-request-dashboard.cy.js)                                │
  │ Searched:       cypress/e2e/add-forge-now-request-dashboard.cy.js                              │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  add-forge-now-request-dashboard.cy.js                                           (1 of 1)


  Test add forge now request dashboard load
    ✓ should load add forge request details
    ✓ should show send message link

...
```

**Comment by romankhomitskyi** [comment]
[cypress-cloud] Could not get cypress attempt error details
so that is the latest error?

**Comment by romankhomitskyi** [comment]
"cypress": "13.2.0",
"cypress-cloud": "2.0.0-beta.0",
<img width="554" alt="Screenshot 2023-09-28 at 18 20 22" src="https://github.com/currents-dev/cypress-cloud/assets/57874507/96dbfc54-b6c1-4698-b5af-33424e4e4fef">
it does not upload screenshots when before all hook is failed

**Comment by jiangzhuo** [comment]
> > This will be hard one, i guess...
> > Seems like this check happens partially remote. If so, its an interesting point that cypress seems to call home with some repo infos 🤡
> > Our Pipeline already worked with cypress 13.2 since some days... so last night this check "has been activated" in some way
> 
> Same issue from my side, regardless `cypress-cloud` version used (tested with `1.10.0-beta.4` and `2.0.0-beta.1`): https://jenkins.softwareheritage.org/blue/organizations/jenkins/DWAPPS%2Fgitlab-builds/detail/gitlab-builds/320/pipeline.
> 
> Curiously, I can run the tests locally without any issue:
> 
> ```
> $ CURRENTS_API_URL=http://localhost:1234 CURRENTS_PROJECT_ID=swh-web yarn run cypress-cloud run --record --key swh-web --parallel --ci-build-id build-2
> yarn run v1.22.19
> $ /home/anlambert/swh/swh-environment/swh-web/node_modules/.bin/cypress-cloud run --record --key swh-web --parallel --ci-build-id build-2
>  WARNING  Failed to load config file, falling back to the default config. Attempted locations: [
>   'file:///home/anlambert/swh/swh-environment/swh-web/currents.config.js',
>   'file:///home/anlambert/swh/swh-environment/swh-web/currents.config.cjs',
>   'file:///home/anlambert/swh/swh-environment/swh-web/currents.config.mjs'
> ]
> 
> Copyright (C) 2023 Currents Software Inc https://currents.dev
> This is free software, and you are welcome to redistribute it under certain
> conditions. This program comes with no warranty. Parts of this program are MIT
> licensed. Refer to the license for details
> https://github.com/currents-dev/cypress-cloud/blob/main/LICENSE.md
> 
> Cypress-cloud version: 2.0.0-beta.1
> Cypress version: 13.3.0
> Discovered 25 spec files
> Tags: false; Group: false; Parallel: true; Batch Size: 3
> Connecting to cloud orchestration service...
> 🎥 Run URL: http://localhost:8080/run/e455e0bdc0d3226e99bc24547f2ea7ab
> 
> ====================================================================================================
> 
> Running: cypress/e2e/add-forge-now-request-dashboard.cy.js (2/25)
> 
> ====================================================================================================
> 
>   (Run Starting)
> 
>   ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
>   │ Cypress:        13.3.0                                                                         │
>   │ Browser:        Electron 114 (headless)                                                        │
>   │ Node Version:   v18.16.0 (/usr/bin/node)                                                       │
>   │ Specs:          1 found (add-forge-now-request-dashboard.cy.js)                                │
>   │ Searched:       cypress/e2e/add-forge-now-request-dashboard.cy.js                              │
>   └────────────────────────────────────────────────────────────────────────────────────────────────┘
> 
> 
> ────────────────────────────────────────────────────────────────────────────────────────────────────
>                                                                                                     
>   Running:  add-forge-now-request-dashboard.cy.js                                           (1 of 1)
> 
> 
>   Test add forge now request dashboard load
>     ✓ should load add forge request details
>     ✓ should show send message link
> 
> ...
> ```

same here, I can run in my local successfully, but in lambda environment(Lambda function from a container image) got the 3rd party error. I also tried to run the docker image in my local with `docker run`, also got the 3rd party error.

**Comment by jiangzhuo** [comment]
Based on my experimental guesses, the logic for detection is in the cypress binary and does not require an internet connection, and there may be two points of detection:
1. Is there `cypress-cloud` in node_modules
2. is there currents.config.js in project root

**Comment by mix4242** [comment]
https://www.cypress.io/blog/2023/09/25/defense-intellectual-property

**Comment by Roemer** [comment]
I am a bit confused. Cypress launched the `Cypress Gateway Connector` License, (costing around 5k$/y) so that a 3rd party recording solution can be used. And now that does not work? I could not test with the latest version of Sorry Cypress but I will get in touch with Cypress to check, if that should work with this license or not.

**Comment by ssurendra549** [comment]
We are getting this error while running cypress in Jenkins pipeline -

"We've detected that you're using a 3rd party library that is not supported by Cypress: cypress-cloud
To continue running Cypress, please remove this library or reach out for help migrating."

below is my configuration:

 "cypress": "^13.3.0",
 "cypress-cloud": "^2.0.0-beta.1"
 
 followed the migration document and made the changes accordingly - https://currents.dev/readme/integration-with-cypress/cypress-cloud/migration-to-cypress-13

**Comment by bilarallen** [comment]
The Cypress team is already investigating and working on a fix https://github.com/cypress-io/cypress/discussions/27924 it seems other libraries that do not make test recordings were also affected. 🤞 that this will solve the issue with currents, if not where do we go from here? 🥲

**Comment by agoldis** [comment]
Hey everyone, we published [the response](https://currents.dev/posts/v13-blocking) with more details about the recent blocking, including the technical analysis. We need to change our approach because they will keep implementing new blockings. It requires careful planning and more time. Thanks for your patience and support!

**Comment by Roemer** [comment]
According to Cypress, they want to get this working for paying customers (like we are with the `Cypress Gateway Connector`) because they themselves do not have an on-premise dashboard. @agoldis did you ever have contact with Cypress directly? I am in contact with them and they need more details in order to fix it.

**Comment by agoldis** [comment]
@Roemer 
- They will offer Cypress Gateway Connector for $400-500/mo for our old integration https://www.npmjs.com/package/cy2
- This will block our new integration https://github.com/currents-dev/cypress-cloud regardless of `Cypress Gateway Connection`
- They won't "fix" the blocking - they introduced it on purpose to block Sorry Cypress and Currents

**Comment by papermana** [comment]
Would forking Cypress be a viable solution to this?

**Comment by Ido112** [comment]
Hello, I found another issue with Cypress 13 and currents, 
I'm using the hook 'test:after:run', (https://docs.cypress.io/api/cypress-api/catalog-of-events#Cypress-Events)
when i use Cypress runner, it works perfectly, when i use the currents runner, its fails with this error message
![image](https://github.com/currents-dev/cypress-cloud/assets/32884355/ce0dd1ea-d232-44ad-bf7f-f1e81069524c)

**Comment by estefafdez** [comment]
> > Would forking Cypress be a viable solution to this?
> 
> I made a fork of cypress-cloud and renamed everything possible. And it worked for.version 13.3.0. I also had to rename all environment variables, for example CURRENTS_API_URL

Could you share how to do it please?

**Comment by Krivega** [comment]
> > > Would forking Cypress be a viable solution to this?
> > 
> > 
> > I made a fork of cypress-cloud and renamed everything possible. And it worked for.version 13.3.0. I also had to rename all environment variables, for example CURRENTS_API_URL
> 
> Could you share how to do it please?


https://www.npmjs.com/package/@krivega/cc

This is a temporary solution, now I am afraid that they will block it too)

**Comment by GeuntaBuwono** [comment]
so it's dead end of 3rd party self hosted cypress monitor ? 😭

**Comment by agoldis** [comment]
@GeuntaBuwono you can still use older versions of cypress https://currents.dev/readme/integration-with-cypress/alternative-cypress-binaries

**Comment by tomasjanicek** [comment]
cypress has already released stable version 14.0.0.. so isn't it time to support the latest version cypress 13 stably?

**Comment by SimonFischer04** [comment]
@tomasjanicek it is not like they don't want to support v13.
They can't support it, because cypress actively blocks them.

**Comment by tomasjanicek** [comment]
Was that how it worked with cypress 12? I understood that the strategy is always to wait for the latest major stable version and then try to make a bypassed package... I'm mainly interested in the strategy. If there is no prospect that this package will be developed, I would like to know so that I can switch to something else.
